### PR TITLE
build ockam on cross platforms `on: pull_request`

### PR DESCRIPTION
**Pull request by metaclips**
Build ockam on cross platform `on: pull_request` to ensure we do not break our releases while merging PRs. Also updates cache version

### CODE REVIEW

File: .github/actions/build_binaries/action.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Build Ockam Binaries For Different Architechtures
+description: Build Ockam Binaries For Different Architechtures
+
+inputs:
+  use_cross_build:
+    description: Indicate If Cross Should Be Used To Build Binary Instead Of Cargo
```
Comment (etorreborre) [inline .github/actions/build_binaries/action.yml:6]: I don't know what the accepted convention but Do We Need To Capitalize Every Word :-)?

File: .github/actions/build_binaries/action.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      run: |
+        set -x
+        use_cross_build="${{ inputs.use_cross_build }}"
+
+        if [[ "${{ inputs.target }}" =~ .+-musl(.+)?  ]]; then
+          export RUSTFLAGS='-C target-feature=+crt-static'
+        fi
+
+        if [[ $use_cross_build == "true" ]]; then
+          cross build --bin ockam --target ${{ inputs.target }} --verbose --release
```
Comment (etorreborre) [inline .github/actions/build_binaries/action.yml:52]: I don't see any difference between the `if` and the `else` part

**Review by etorreborre — APPROVED** [review]
Except the if/else thing I don't see anything wrong (but I'm not an expert in Github actions :-))

File: .github/actions/build_binaries/action.yml (hunk 2 above)
Comment (metaclips) [reply .github/actions/build_binaries/action.yml:52]: One uses [cross](https://github.com/cross-rs/cross) to build ockam command and the else part uses conventional cargo command

File: .github/actions/build_binaries/action.yml (hunk 1 above)
Comment (metaclips) [reply .github/actions/build_binaries/action.yml:6]: I don't think there's a convention, its more of preference of mine.

**Pull request merged**